File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.cpp
class Solution {
public:
    int evalRPN(vector<string>& tokens)
    {
        stack<int> _stack;

        for(int i = 0;i < tokens.size();i++)
        {
            if (tokens[i] == "+" || tokens[i] == "/" || tokens[i] == "*" || tokens[i] == "-")
            {
            int b = _stack.top();
            _stack.pop();
            int a = _stack.top();
            _stack.pop();
            if (tokens[i] == "+")
                _stack.push(a + b);
            else if (tokens[i] == "-")
                _stack.push(a - b);
            else if (tokens[i] == "*")
                _stack.push(a * b);
            else if (tokens[i] == "/")
                _stack.push(a / b);
            }
            else
            {
                stringstream ss(tokens[i]);
                int num;
                ss >> num;
                _stack.push(num);
            }
        }
        return (_stack.top());
        
    }
};

File: 150-evaluate-reverse-polish-notation/evaluate-reverse-polish-notation.py
class Solution:
    def evalRPN(self, tokens: List[str]) -> int:
        number = []
        operator = ['+', '*', '-', '/']
        c = int(tokens[0])
        for i in tokens:
            if i in operator:
                a = int(number.pop())
                b = int(number.pop())
                if i == '+':
                    c = b + a
                elif i == '-':
                    c = b - a
                elif i == '*':
                    c = b * a
                elif i == '/':
                    c = int(b / a)
                number.append(c)
            else :
                number.append(i)
        return (c)